include/Core/Config.php

```diff
@@ -132,15 +132,33 @@ public static function set($family,$key,$value) {
 		$dbvalue = (is_array($value)?serialize($value):$value);
 		$dbvalue = (is_bool($dbvalue) ? intval($dbvalue) : $dbvalue);
 
-		$ret = q("INSERT INTO `config` ( `cat`, `k`, `v` ) VALUES ( '%s', '%s', '%s' )
-ON DUPLICATE KEY UPDATE `v` = '%s'",
+		// The "INSERT" command is very cost intense. It saves performance to do it this way.
```

Comment:
> The "save_timestamp" is used by the "rendertime" addon which is used to detect code that needs to be optimized.

Yes, I saw that. You'll just have to had `$a->config['system']['profiler'] = true;` to your `.htconfig.php` file to allow your addon to do its magic.

> Problem is that there are many systems out there that are running on small systems with bad I/O (like on shared hosters)

In that case the combination between our two solutions would be the best: less DB queries (= disk I/O) overall, and no systematic INSERT try.

> What do you say to "REPLACE INTO"? It is not SQL standard but is said to be faster.

It definitely would, however I went with `INSERT ... ON DUPLICATE KEY ...` because I've been told that we might have to support PostGreSQL at some point in the future.